### ISSUE TITLE
False positive for NewType name wrapped in Annotation

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#7645]
pyright 1.1.357 can catch invalid NewType names

```python
MyIds = NewType("WrongName", str) # reportGeneralTypeIssues
```

but if i wrap NewType with Annotated

```python
MyId = Annotated[NewType("WrongName", str), Doc("help")]
```

he fails, no errors was shown.

Also I'm not sure if using this syntax is legal or not.

**Comment by erictraut** [comment]
Pyright is working correctly here. When you wrap a `NewType` with `Annotated`, the resulting object is no longer a `NewType`. It's a type alias. It's name does not need to match that of the (anonymous) NewType that was created.
